src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -693,7 +693,7 @@ public override void GetDependenciesDueToAccess(ref DependencyList dependencies,
             string reason = "Use of a field";
 
             bool generatesMetadata = false;
-            if (!IsReflectionBlocked(writtenField))
+            if (!writtenField.OwningType.IsRuntimeDeterminedSubtype && !IsReflectionBlocked(writtenField))
```

Comment:
It's usually a bug if runtime determined forms reach here. Someone should have converted to canonical. I think we might miss things if we treat these as blocked.

Reply:
Thanks for pointing this out - fixed by passing only canonical forms from the ILScanner.